src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/corporate/CorporateMappingResourceIntTest.kt

```diff
@@ -0,0 +1,772 @@
+        webTestClient.post()
+          .uri("/mapping/corporate/migrate")
+          .headers(setAuthorisation(roles = listOf("NOMIS_CONTACTPERSONS")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            BodyInserters.fromValue(
+              mappings.copy(
+                corporateEmailMapping = listOf(
+                  CorporateMappingIdDto(dpsId = "11", nomisId = 1),
+                  CorporateMappingIdDto(dpsId = "22", nomisId = 2),
+                ),
+              ),
+            ),
+          )
+          .exchange()
+          .expectStatus().isCreated
+
+        with(corporateEmailMappingRepository.findOneByNomisId(1)!!) {
+          assertThat(dpsId).isEqualTo("11")
+          assertThat(nomisId).isEqualTo(1L)
+          assertThat(label).isEqualTo(mappings.label)
+          assertThat(mappingType).isEqualTo(mappings.mappingType)
+          assertThat(whenCreated).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
+        }
+        with(corporateEmailMappingRepository.findOneByDpsId("22")!!) {
+          assertThat(dpsId).isEqualTo("22")
+          assertThat(nomisId).isEqualTo(2L)
+          assertThat(label).isEqualTo(mappings.label)
+          assertThat(mappingType).isEqualTo(mappings.mappingType)
+          assertThat(whenCreated).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
+        }
+      }
+
+      @Test
+      fun `will persist the corporate web mapping`() = runTest {
+        webTestClient.post()
+          .uri("/mapping/corporate/migrate")
+          .headers(setAuthorisation(roles = listOf("NOMIS_CONTACTPERSONS")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            BodyInserters.fromValue(
+              mappings.copy(
+                corporateWebMapping = listOf(
+                  CorporateMappingIdDto(dpsId = "11", nomisId = 1),
+                  CorporateMappingIdDto(dpsId = "22", nomisId = 2),
+                ),
+              ),
+            ),
+          )
+          .exchange()
+          .expectStatus().isCreated
+
+        with(corporateWebMappingRepository.findOneByNomisId(1)!!) {
+          assertThat(dpsId).isEqualTo("11")
+          assertThat(nomisId).isEqualTo(1L)
+          assertThat(label).isEqualTo(mappings.label)
+          assertThat(mappingType).isEqualTo(mappings.mappingType)
+          assertThat(whenCreated).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
+        }
+        with(corporateWebMappingRepository.findOneByDpsId("22")!!) {
+          assertThat(dpsId).isEqualTo("22")
+          assertThat(nomisId).isEqualTo(2L)
+          assertThat(label).isEqualTo(mappings.label)
+          assertThat(mappingType).isEqualTo(mappings.mappingType)
+          assertThat(whenCreated).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
+        }
+      }
+    }
+  }
+
+  @DisplayName("GET /mapping/corporate/corporate/migration-id/{migrationId}")
```

Comment:
Looks weird /corporate/corporate but because corporate is a sub domain within corporates - might rethink this